vars/branchRepo.groovy

```diff
@@ -1,42 +1,39 @@
 #!/usr/bin/groovy
 
-def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body = {}) {
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, boolean useExistingBranch = false, Closure body = {}) {
```

Comment:
New optional argument, if true will use any exiting branch on the repo if it already exists, false it does the usual checks.